ethdk/src/ExternallyOwnedAccount/ExternallyOwnedAccount.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers, Wallet } from 'ethers'
+import { type Deferrable } from 'ethers/lib/utils'
+import type Account from '../interfaces/Account'
+import {
+  type ExternallyOwnedAccountNetwork,
+  type Network,
+} from '../interfaces/Network'
+import type Transaction from '../interfaces/Transaction'
+import { getNetwork } from './ExternallyOwnedAccountNetworks'
+import ExternallyOwnedAccountTransaction from './ExternallyOwnedAccountTransaction'
+
+export default class ExternallyOwnedAccount implements Account {
+  accountType: string = 'eoa'
+
+  address: string
+  private readonly privateKey: string
+  private readonly recoveryPhrase: string
+  private readonly networkConfig: ExternallyOwnedAccountNetwork
+  private readonly provider: ethers.providers.JsonRpcProvider
+  private readonly signer: Wallet
+
+  private constructor({
+    address,
+    privateKey,
+    recoveryPhrase,
+    network,
+    provider,
+    signer,
+  }: {
+    address: string
+    privateKey: string
+    recoveryPhrase: string
+    network: ExternallyOwnedAccountNetwork
+    provider: ethers.providers.JsonRpcProvider
+    signer: Wallet
+  }) {
+    this.address = address
+    this.privateKey = privateKey
+    this.recoveryPhrase = recoveryPhrase
+    this.networkConfig = network
+    this.provider = provider
+    this.signer = signer
+  }
+
+  static async createAccount({
+    privateKey,
+    network,
+  }: {
+    privateKey?: string
+    network?: Network
+  } = {}): Promise<ExternallyOwnedAccount> {
+    const signer = Wallet.createRandom(privateKey)
+
+    const networkConfig = getNetwork(network)
+
+    const provider = new ethers.providers.JsonRpcProvider(
+      networkConfig.rpcUrl,
+      {
+        name: networkConfig.name,
+        chainId: networkConfig.chainId,
+      },
+    )
+
+    return new ExternallyOwnedAccount({
+      address: signer.address,
+      privateKey: signer.privateKey,
+      recoveryPhrase: signer.mnemonic.phrase,
```

Comment:
Curious to hear your reasoning for naming the seed phrase a `recoveryPhrase`?  I'm not necessarily against it, I just think of a mnemonic and the private key similarly.  I think they are more for generating a wallet.  Not necessarily for recovering a wallet that exists.

Reply:
Good point, I think I was just thinking about recovery and didn't consider wallet generation. My reasoning was that recoveryPhrase then seemed more helpful (less web3 jargon-y) over mnemonic or seedPhrase. But I agree the mnemonic or seedPhrase better encapsulates the recovery and generation aspect. Will update